### PR TITLE
Subreddit form

### PR DESCRIPTION
**Pull request by dcwds**
The following changes have been made according to the AC:
- add Search component that includes: title, text input (`defaultValue` is set to search url param) prefixed with `r/`, search button
- add useSearchForm hook to manage state, handlers, and history
- add tests for useSearchForm hook

Other changes:
- add `@testing-library/react-hooks` to deps
- update `jsx-a11y` rule in ESLint to allow styled components to work well with labels

### CODE REVIEW

**Comment by mentor-tara** [comment]
Great job :rocket: I'm a bit busy at the moment but I'll leave a review soon.

In the meantime, consider starting with the next task already to practice the professional Git workflow. [If you need a refresher, feel free to follow this section in the course again.](https://profy.dev/project/reddit-timer/while-waiting)[](ckmadjf0g3nm60a84ilt0ydvy)

**Review by mentor-tara — CHANGES_REQUESTED** [review]
Before I approve this PR I'd like to share a few tips with you.

**Please go through the list below and review your own code first.** Imagine you're the reviewer or a person who didn't see the code before. Ask yourself if everything is clearly written and understandable. Are components and variables properly named?

Reviewing your own code with this mindset is an important practice that will help you learn a lot and write high-quality code.

To make it a bit easier for you here are the tips I promised. Some have a link :link: attached where you can find more information. I'd recommend reading the **bold tips** in any case. Feel free to check the boxes when you think your code fits.

**General:**
- [ ] Use controlled form inputs [:link:](https://profy.dev/project/reddit-timer/review-tips/controlled-inputs)
- [ ] Use URL params instead of query params [:link:](https://profy.dev/project/reddit-timer/review-tips/prefer-url-params)
- [ ] Don't access the DOM directly e.g. via `document.getElementById` [:link:](https://profy.dev/project/reddit-timer/review-tips/accessing-dom-directly)
- [ ] Don't mutate function parameters [:link:](https://profy.dev/project/reddit-timer/review-tips/mutating-function-params)
- [ ] **Make sure variable names are descriptive** [:link:](https://profy.dev/project/reddit-timer/review-tips/variable-naming)
- [ ] Rename URL param locally if required using descructuring [:link:](https://profy.dev/project/reddit-timer/review-tips/rename-variable-via-destructuring)
- [ ] **Use a library for handling forms (optional)** [:link:](https://profy.dev/project/reddit-timer/review-tips/form-library)
- [ ] [useHistory](https://reacttraining.com/react-router/web/api/Hooks/usehistory) hook to update the route

**Styles:**
- [ ] Extract button as shared component between Hero Section and Subreddit Form [:link:](https://profy.dev/project/reddit-timer/review-tips/button-as-shared-compoent)
- [ ] Only set a fixed width to the input. All other widths can be auto
- [ ] Use 10px as margin between form elements
- [ ] Using `justify-content` to position form elements is not necessary
- [ ] **Refrain from styling children with CSS selectors** [:link:](https://profy.dev/project/reddit-timer/review-tips/css-selectors)
- [ ] Use global styles when appropriate (e.g. for font-size of headings, color of links...) [:link:](https://profy.dev/project/reddit-timer/review-tips/globally-style-elements)

Once you're done with reviewing your own code feel free to check the [example implementation](https://profy.dev/project/reddit-timer/code-examples/subreddit-form-ca02d443-5d06-4d52-b3ca-9eab5cea6858). **But remember: It's an important exercise to review your code first on your own.**

When you think this PR is ready to be merged just let me know by requesting another review. I'll approve right away.[](ckmcyyyz408ua0b28fmlwcds4)

File: src/hooks/use-subreddit-form/use-subreddit-form.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react';
+
+const useSubredditForm = (initialSubreddit, history) => {
+  const [subreddit, setSubreddit] = useState(initialSubreddit);
+
+  const changeSubreddit = (e) => setSubreddit(e.target.value);
+
+  const searchSubreddit = (e) => {
```
Comment (jkettmann) [inline src/hooks/use-subreddit-form/use-subreddit-form.js:8]: Great that you detected the unnecessary useEffect. Makes the state also much simpler.

File: src/hooks/use-subreddit-form/use-subreddit-form.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react';
+
+const useSubredditForm = (initialSubreddit, history) => {
+  const [subreddit, setSubreddit] = useState(initialSubreddit);
```
Comment (jkettmann) [inline src/hooks/use-subreddit-form/use-subreddit-form.js:4]: I'd move the `useHistory` and `useParams` hooks also here I guess.

**Review by mentor-tara — APPROVED** [review]
Great work! :tada::tada::tada: You can merge the PR now.[](ckmadnlh43o2n0b87e55s83qv)

**Pull request merged**